codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HpackUtil.java

```diff
@@ -39,8 +42,16 @@ static boolean equals(CharSequence s1, CharSequence s2) {
         if (s1.length() != s2.length()) {
             return false;
         }
+
+        if (s1 instanceof AsciiString && s2 instanceof AsciiString) {
+            AsciiString s1Ascii = (AsciiString) s1;
+            AsciiString s2Ascii = (AsciiString) s2;
+            return PlatformDependent.equalsConsistentTime(s1Ascii.array(), s1Ascii.arrayOffset(),
+                                                          s2Ascii.array(), s2Ascii.arrayOffset(), s1.length());
+        }
+
         char c = 0;
```

Comment:
@carl-mastrangelo - I repeated your benchmark when charsequence is used and not surprisingly I see the same results ... i'll make the same change here too

```
Benchmark                  Mode     Cnt     Score   Error  Units
TestBenchmark.theChars   sample  152044  2105.563 ± 7.328  ns/op
TestBenchmark.theInts    sample  141490  1155.404 ± 6.013  ns/op
TestBenchmark.theLongs   sample  108386  1498.885 ± 8.700  ns/op
TestBenchmark.theShorts  sample  151317  2119.505 ± 8.116  ns/op
```